src/test/java/seedu/address/logic/commands/DoneTaskCommandTest.java

```diff
@@ -0,0 +1,124 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.logic.commands.CommandTestUtil.showTaskAtIndex;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_TASK;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_TASK;
+import static seedu.address.testutil.TypicalIndexes.INDEX_THIRD_TASK;
+import static seedu.address.testutil.TypicalTasks.getTypicalAddressBook;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.task.Task;
+import seedu.address.testutil.TaskBuilder;
+
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code DoneTaskCommand}.
+ */
+public class DoneTaskCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_validIndexUnfilteredList_success() {
+        Task taskToMarkDone = model.getFilteredTaskList().get(INDEX_FIRST_TASK.getZeroBased());
+
+        Task doneTask = new TaskBuilder(taskToMarkDone).withTaskStatus(TaskBuilder.COMPLETED_STATUS).build();
+
+        DoneTaskCommand doneCommand = new DoneTaskCommand(INDEX_FIRST_TASK);
+
+        String expectedMessage = DoneTaskCommand.MESSAGE_DONE_TASK_SUCCESS;
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
```

Comment:
For expectedModel, would it be better to declare it as a class instance variable instead?